### DISCUSSION TITLE
Use filter :length with back relations

### DISCUSSION THREAD

**Discussion by pnmcosta** [pocketbase/pocketbase#4499] [Q&A] [answered]
Hi,

I don't think this is currently possible, as I've tried it and got an error on the collection list route.

I'm currently using a field within the back relation result `id` to check whether it has back relation records or not, but I now have a use case where I'd like to distinguish between the number of existing back relations.

My request is being done with the JS sdk to a collection called `members`, there's another collection `rewards` with a field that references the `member` on the `members` collection:

```
    collectionName = "members"
    options = {
        expand: "rewards_via_member",
        fields: "id,expand.rewards_via_member.id"
    }
```

Using
```
filter = `rewards_via_member.id!=""`
```
works and I'm only returned `members` with rewards.

But I'd like to use
```
filter = `rewards_via_member:length>0`
```

Or
```
filter = `rewards_via_member:length=3`
```

Which doesn't seem to work, I can't recall the error now, but I can reproduce it if needed.

Any workaround/suggestion would be great.

Thank you

**Comment by ganigeorgiev** [comment, marked as answer]
`:length` works only with "array-able" fields, aka. fields where the raw value could be represented as json array (internally it uses the SQLite [`json_array_length`](https://www.sqlite.org/json1.html#jarraylen)).

Back-relations are different in this regard and if we have to add support for it we'll need to perform a separate `COUNT` sub query only for the joined back-related records (_I've added it in the roadmap to consider it in the future_).

As a workaround, you can create a helper View collection that will contain your count per member, something like:

```sql
SELECT
    members.id,
    members.id as member,
    count(rewards.id) as totalRewards
FROM members
LEFT JOIN rewards ON rewards.member = members.id
GROUP BY members.id

-- OR directly from the rewards (will not have count for missing members relation)

SELECT
    rewards.id,
    rewards.member,
    count(*) as totalRewards
FROM rewards
GROUP BY rewards.member
```

And then you can join it similarly in your filter with something like:

```js
rewardsView_via_member.totalRewards ?= 3
```

_edit: added an alternative sample view query_

**Reply by ganigeorgiev** [reply]
_Edit2: updated to use `?=` because back-relations are evaluated by default as multiple relation field (see [Back-relation caveats](https://pocketbase.io/docs/working-with-relations/#back-relation-caveats)). Both will work in this particular case, but the `?=` will be slightly more efficient as it will skip the extra match-all query check._

**Reply by pnmcosta** [reply]
Yup, worked a treat, thanks for the help. Fyi, I actually used `?>` operand successfully.
